include/coreir-c/coreir.h

```diff
@@ -215,6 +215,7 @@ extern void COREModuleAddMetaDataStr(
   char* value);
 
 // This returns a malloc'd string that should be freed by the user
+extern const char* COREGeneratorGetMetaData(COREGenerator* module);
 extern const char* COREModuleGetMetaData(COREModule* module);
 extern const char* COREInstanceGetMetaData(COREWireable* module);
```

Comment:
Can we collapse these 3 into a single function since they all inherit from the `Metadata` base class?

Reply:
It seems like that should be possible. Specifying the types can get a little annoying though. There is no current method to 'cast' from a COREModule* to a COREMetadata*